client/board_test.go

```diff
@@ -143,3 +143,109 @@ func TestClassicBoards(t *testing.T) {
 		assert.True(t, de.IsNotFound())
 	})
 }
+
+func TestFlexibleBoards(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	var flexibleBoard *client.Board
+	var err error
+
+	c := newTestClient(t)
+	dataset := testDataset(t)
+
+	column, err := c.Columns.Create(ctx, dataset, &client.Column{
+		KeyName: test.RandomStringWithPrefix("test.", 8),
+	})
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		c.Columns.Delete(ctx, dataset, column.ID)
+	})
+
+	query, err := c.Queries.Create(ctx, dataset, &client.QuerySpec{
+		Calculations: []client.CalculationSpec{
+			{
+				Op:     client.CalculationOpAvg,
+				Column: &column.KeyName,
+			},
+		},
+		TimeRange: client.ToPtr(3600), // 1 hour
+	})
+	require.NoError(t, err)
+	require.NotEmpty(t, query.ID)
+
+	qa := &client.QueryAnnotation{
+		Name:        test.RandomStringWithPrefix("test.", 20),
+		Description: "This derived column is created by a test",
+		QueryID:     *query.ID,
+	}
+	queryAnnotation, err := c.QueryAnnotations.Create(ctx, dataset, qa)
+	require.NoError(t, err)
+	require.NotNil(t, queryAnnotation.ID)
+
+	sli, err := c.DerivedColumns.Create(ctx, dataset, &client.DerivedColumn{
+		Alias:      test.RandomStringWithPrefix("test.", 8),
+		Expression: "BOOL(1)",
+	})
+	require.NoError(t, err)
+	slo, err := c.SLOs.Create(ctx, dataset, &client.SLO{
+		Name:             test.RandomStringWithPrefix("test.", 8),
+		TimePeriodDays:   7,
+		TargetPerMillion: 990000,
+		SLI:              client.SLIRef{Alias: sli.Alias},
+	})
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		c.SLOs.Delete(ctx, dataset, slo.ID)
+		c.DerivedColumns.Delete(ctx, dataset, sli.ID)
+	})
+
+	t.Run("Create flexible board", func(t *testing.T) {
```

Comment:
Update flexible boards will come as a Separate test since the backend on `main` branch doesn't support that yet.